src/openassetio-python/tests/package/hostApi/test_manager.py

```diff
@@ -3366,6 +2776,246 @@ def test_when_empty_then_no_state_and_stateFromPersistenceToken_is_not_called(
         mock_manager_interface.mock.stateFromPersistenceToken.assert_not_called()
 
 
+batch_element_error_codes_names = [
+    "unknown",
+    "invalidEntityReference",
+    "malformedEntityReference",
+    "entityAccessError",
+    "entityResolutionError",
+    "invalidPreflightHint",
+    "invalidTraitSet",
+]
+
+
+def make_expected_err_msg(batch_element_error, index, access, entityRef):
+    error_type_name = batch_element_error_codes_names[
+        batch_element_error_codes.index(batch_element_error.code)
+    ]
+    return f"{error_type_name}: {batch_element_error.message} [index={index}] [access={kAccessNames[access]}] [entity={entityRef}]"
+
+
+# Conveniences to allow us to paramaterize singular and batch
+def ensure_list(item):
+    return item if isinstance(item, list) else [item]
+
+
+def ensure_singular(item):
+    return item[0] if isinstance(item, list) else item
+
+
+class Test_Manager_convenience_exceptions:
+    @pytest.mark.parametrize("error_code", batch_element_error_codes)
+    @pytest.mark.parametrize("access", [access.ResolveAccess.kRead, access.ResolveAccess.kWrite])
+    @pytest.mark.parametrize("singular", [True, False])
+    def test_when_batch_resolve_emits_BatchElementError_then_appropriate_exception_raised(
+        self,
+        manager,
+        mock_manager_interface,
+        a_host_session,
+        some_refs,
+        an_entity_trait_set,
+        a_context,
+        error_code,
+        access,
+        singular,
+        invoke_resolve_error_cb,
+    ):
+        method = mock_manager_interface.mock.resolve
+
+        expected_index = 0
+
+        # We're parametrizing over different signatures here for
+        # brevity, we want to invoke different overloads, either the
+        # singular convenience or the batch method.
+        singular_or_multi_ref = some_refs[0] if singular else some_refs
+
+        batch_element_error = BatchElementError(error_code, "some string ✨")
+
+        def call_callbacks(*_args):
+            invoke_resolve_error_cb(expected_index, batch_element_error)
+            pytest.fail("Exception should have short-circuited this")
+
+        method.side_effect = call_callbacks
+
+        with pytest.raises(
+            BatchElementException,
+            match=re.escape(
+                make_expected_err_msg(
+                    batch_element_error,
+                    expected_index,
+                    access,
+                    ensure_singular(singular_or_multi_ref),
+                )
+            ),
+        ) as exc:
+            manager.resolve(
+                singular_or_multi_ref,
+                an_entity_trait_set,
+                access,
+                a_context,
+                Manager.BatchElementErrorPolicyTag.kException,
```

Comment:
[fix] We've lost coverage of the fact that the default signature is the exception signature.  

Perhaps yet another parametrization like `"tag",  [[], [kException]]` then passed as `*tag`?

Reply:
Oh right! That's why we were parameterizing over those tags, I thought some prior tags had just made a mistake, took it out on purpose.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1109/commits/7a0d88222e3288bbe2b7e70777ae718975a351de

Reply:
Could we add the same parametrization to the other tests too?

Reply:
I'm a moron ... yes.

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/1109/commits/d90d160c0fc9b33a78f83a47eca6fd0fe207e93e